packages/api-react/src/hooks/usePrefs.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useEffect, useState } from 'react';
+import EventEmitter from '../utils/EventEmitter';
+
+const eventEmitter = new EventEmitter();
+
+function deepEqual(objA: any, objB: any){
+  return JSON.stringify(objA) === JSON.stringify(objB);
```

Comment:
I believe that the keys in an object aren't guaranteed to be ordered in any deterministic way. So `{ foo: 1, bar: 2 }` and `{ bar: 2, foo: 1 }` may not be equal.

Looking at how this function is used (currently), I don't think it's a problem, we would just end up setting the new value even though semantically the old and new values are equivalent.

It would probably be worth adding a comment at the very least to explain that this isn't a very robust way to test for object equality so that we don't accidentally use this type of comparison elsewhere in the code.

Reply:
Makes sense.
I renamed it to `maybeEqual`.
I think it doesn't need to strictly check equality with execution cost because this check was introduced for enhancing performance.